### PR TITLE
Add apidocs by default to /

### PR DESCRIPTION
**Pull request by dshafik**
This change adds a default / route, and HTML template, that will render nicely formatted documentation automagically.

The documentation is taken from the config data, and it will render action/error descriptions using markdown.

The "Introduction" section is simply the description for the '/' route itself, and is handled specially.

You can see what this looks like by default, here: http://screencast.com/t/jHfaSjTd

### CODE REVIEW

**Comment by davidcoallier** [comment]
Going over this, it's looking pretty spiffy. We'll need to update http://frapi.github.com accordingly as well to reflect that.

**Comment by dshafik** [comment]
Added docs to frapi.github.com (per @davidcoallier) in frapi/frapi.github.com#4

File: src/frapi/custom/Output/html/Introduction.html.tpl [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+                                                            echo isset($subparam['required']) && $subparam['required'] == '1'
+                                                                ? '<strong>&#10003;</strong>' : '<strong>&#10007;</strong>'
+                                                        ?>
+                                                    </td>
+                                                </tr>
+                                                <?php
+                                            }
+                                        }
+                                        ?>
+                                        <tr>
```
Comment (clayhinson) [inline src/frapi/custom/Output/html/Introduction.html.tpl:308]: I don't think this section (307-317) should be here, it generates an error rendering the template. if params exist they will be output in 293-303.

Comment (davidcoallier) [inline src/frapi/custom/Output/html/Introduction.html.tpl:308]: This is from the original documentation generator. We'll need to resolve this.

**Pull request merged**